### PR TITLE
Rework DomU configs

### PR DESCRIPTION
**Pull request by rshym**
* Change MAC for DomU domain
* remove not needed executable attribute from config file

### CODE REVIEW

**Comment by rshym** [comment]
According to offline discussion:
* do not use domF's MAC but assign dedicated MAC and IP
* use dedicated port for ssh to domU
* do not use 2023 as already used for ssh to dom0
* need to test with network boot

**Comment by otyshchenko1** [comment]
both patches look good, feel free to add my R-b after check whether it remains functional with NFS root.

**Comment by rshym** [comment]
Proposal after one more offline discussion - move common part (network settings block) to separate file and add it to configs during build.

**Comment by rshym** [comment]
All comments are addressed:
- disk, vgsx, vif, vdispl, vkb and vsnd moved to separate file
- domU has dedicated MAC, IP and port 2025 for SSH (5 because it's IP is .5)
- added placeholder for port 2023 for SSH to dom0
Also
- three equal domu-*.cfg are replaced by one domu-generic-h3.cfg
- minor changes in formats and comments

Not tested with network boot.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb [hunk 1]
```diff
@@ -9,24 +9,22 @@ LIC_FILES_CHKSUM = "file://${COREBASE}/meta/COPYING.MIT;md5=3da9cfbcb788c80a0384
...
     file://domu-salvator-xs-m3n.cfg \
     file://domu-salvator-x-h3-4x2g.cfg \
     file://guest_domu \
 "
 
 S = "${WORKDIR}"
 
-DOMU_CONFIG_salvator-x-m3-xt = "domu-salvator-x-m3.cfg"
-DOMU_CONFIG_salvator-x-h3-xt = "domu-salvator-x-h3.cfg"
+DOMU_CONFIG_salvator-x-m3-xt = "domu-generic-h3.cfg"
```
Comment (otyshchenko1) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb]: Could we choose some name without "h3" here? xxx-m3 = yyy-h3 sounds a bit confusing.

Comment (otyshchenko1) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb]: Could we choose some name without "h3" here? xxx-m3 = yyy-h3 sounds a bit confusing.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# =====================================================================
+# DomU virtual devices
+# =====================================================================
+
```
Comment (otyshchenko1) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg]: There was a comment here "# Uncomment this when using block device as a boot device" before moving

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb (hunk 1 above)
Comment (rshym) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb]: I had same thoughts but when I noted that 'domu-salvator-x-h3-4x2g.cfg' is used for xs and h3ulcb, I decided to use '-h3' to show that m3 and h3 are equal from this point of view.
But I really OK with any name if you can propose something better.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg (hunk 2 above)
Comment (rshym) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg]: I remove it intentionally because as I see line after "Uncomment" is already uncommented. If you think it's important, I will rephrase it to something like "if .. comment else .. uncomment"

Comment (otyshchenko1) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg]: Whether "disk = ..." string should be (un)commented really depends on "extra = ... " string above. 

If nfs is used as a boot device then "disk = ..." should be commented, but if block device (emmc, sd, whatever) is used as a boot device then "disk = ..." should be **un**commented.

**Comment by otyshchenko1** [comment]
Running ahead, will we likely want the some equivalent of domu-vdevices.cfg for DomA...

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg (hunk 2 above)
Comment (rshym) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domu-vdevices.cfg]: Done.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb (hunk 1 above)
Comment (rshym) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/domu.bb]: Names of generic configs are changed according to offline discussion:
- domu-generic-h3
- domu-generic-h3-4x2g

domu-salvator-x-m3 preserved for future modifications.

**Comment by rshym** [comment]
Comments are addressed.
Suggesters are added.
Not tested yet.

**Comment by otyshchenko1** [comment]
Great work, thank you! LGTM

**Comment by rshym** [comment]
> Running ahead, we will likely want the some equivalent of domu-vdevices.cfg for DomA...

Agree. I have added this to 'todo-list'. Also need to review applicability to tu/aos to keep products aligned.

**Comment by rshym** [comment]
Tested by build of devel-domu-x8.
- dom0  domu.cfg OK
- domd  dnsmasq.conf OK
- domd  port-forward*  OK

Works as expected.
